File: Quotes.c
#include <SDL2/SDL.h>
#include <SDL2/SDL_ttf.h>
#include <stdlib.h>
#include <time.h>

// Daftar quotes
const char *quotes[] = {
    "Hidup ini adalah pilihan.",
    "Kegagalan adalah awal dari kesuksesan.",
    "Jangan pernah berhenti belajar.",
    "Waktu adalah hal terberharga.",
    "Keberanian bukan berarti tidak ada rasa takut, tapi melangkah meski takut.",
    "Mimpi tidak akan berarti apa-apa jika tidak diikuti dengan usaha."
};

#define NUM_QUOTES 6
#define WIDTH 800
#define HEIGHT 400

int main() {
    // Inisialisasi SDL
    if (SDL_Init(SDL_INIT_VIDEO) < 0) {
        printf("SDL gagal diinisialisasi: %s\n", SDL_GetError());
        return -1;
    }

    // Inisialisasi SDL_ttf untuk font
    if (TTF_Init() == -1) {
        printf("SDL_ttf gagal diinisialisasi: %s\n", TTF_GetError());
        return -1;
    }

    // Membuka jendela SDL
    SDL_Window *window = SDL_CreateWindow("Quote Maker", SDL_WINDOWPOS_UNDEFINED, SDL_WINDOWPOS_UNDEFINED, WIDTH, HEIGHT, SDL_WINDOW_SHOWN);
    if (!window) {
        printf("Gagal membuka jendela: %s\n", SDL_GetError());
        return -1;
    }

    // Membuat renderer
    SDL_Renderer *renderer = SDL_CreateRenderer(window, -1, SDL_RENDERER_ACCELERATED);
    if (!renderer) {
        printf("Gagal membuat renderer: %s\n", SDL_GetError());
        return -1;
    }

    // Memilih font
    TTF_Font *font = TTF_OpenFont("arial.ttf", 40);
    if (!font) {
        printf("Gagal membuka font: %s\n", TTF_GetError());
        return -1;
    }

    // Memilih quote acak
    srand(time(NULL));
    const char *quote = quotes[rand() % NUM_QUOTES];

    // Membuat surface dan texture untuk gambar
    SDL_Surface *surface = TTF_RenderText_Blended(font, quote, (SDL_Color){0, 0, 0});
    SDL_Texture *texture = SDL_CreateTextureFromSurface(renderer, surface);

    // Mendapatkan ukuran teks
    int text_width = surface->w;
    int text_height = surface->h;
    SDL_FreeSurface(surface);

    // Menyusun posisi teks di tengah
    SDL_Rect dstrect = { (WIDTH - text_width) / 2, (HEIGHT - text_height) / 2, text_width, text_height };

    // Menampilkan background (cream)
    SDL_SetRenderDrawColor(renderer, 255, 223, 186, 255); // Background cream
    SDL_RenderFillRect(renderer, NULL);

    // Menampilkan teks
    SDL_RenderCopy(renderer, texture, NULL, &dstrect);
    SDL_RenderPresent(renderer);

    // Menunggu sebelum keluar
    SDL_Delay(5000);

    // Bersihkan dan tutup SDL
    SDL_DestroyTexture(texture);
    TTF_CloseFont(font);
    SDL_DestroyRenderer(renderer);
    SDL_DestroyWindow(window);
    TTF_Quit();
    SDL_Quit();

    return 0;
}

File: Quotes.py
from PIL import Image, ImageDraw, ImageFont
import random

# Daftar quotes default yang bisa dipilih
default_quotes = [
    "Hidup ini adalah pilihan.",
    "Kegagalan adalah awal dari kesuksesan.",
    "Jangan pernah berhenti belajar.",
    "Waktu adalah hal terberharga.",
    "Keberanian bukan berarti tidak ada rasa takut, tapi melangkah meski takut.",
    "Mimpi tidak akan berarti apa-apa jika tidak diikuti dengan usaha."
]

# Fungsi untuk membuat gambar dengan quote
def generate_quote_image(quote, output_path='quote_image.png'):
    # Ukuran gambar
    width, height = 800, 400
    background_color = (255, 223, 186)  # Warna background cream

    # Membuat gambar baru dengan ukuran dan warna background
    img = Image.new('RGB', (width, height), color=background_color)

    # Menambahkan teks ke gambar
    draw = ImageDraw.Draw(img)

    # Menentukan font dan ukuran
    try:
        font = ImageFont.truetype("arial.ttf", 40)  # Pastikan fontnya ada
    except IOError:
        font = ImageFont.load_default()

    # Menghitung ukuran teks untuk penempatan tengah
    text_bbox = draw.textbbox((0, 0), quote, font=font)
    text_width = text_bbox[2] - text_bbox[0]  # Lebar teks
    text_height = text_bbox[3] - text_bbox[1]  # Tinggi teks

    # Menghitung posisi teks agar berada di tengah gambar
    position = ((width - text_width) // 2, (height - text_height) // 2)

    # Menambahkan teks ke gambar
    draw.text(position, quote, fill=(0, 0, 0), font=font)

    # Menyimpan gambar ke file
    img.save(output_path)
    print(f"Quote berhasil dibuat dan disimpan di {output_path}")

# Fungsi untuk memilih quote secara manual atau acak
def get_quote_input():
    print("Pilih salah satu opsi:")
    print("1. Masukkan quote sendiri")
    print("2. Pilih quote secara acak")
    choice = input("Masukkan pilihan (1 atau 2): ")

    if choice == "1":
        # Input quote manual dari pengguna
        quote = input("Masukkan quote Anda: ")
    elif choice == "2":
        # Pilih quote acak dari daftar
        quote = random.choice(default_quotes)
        print(f"Quote yang dipilih secara acak: {quote}")
    else:
        print("Pilihan tidak valid! Menggunakan quote acak.")
        quote = random.choice(default_quotes)

    return quote

# Fungsi utama untuk membuat gambar dengan quote
def generate_random_quote_image():
    quote = get_quote_input()  # Mendapatkan quote dari pengguna
    generate_quote_image(quote)  # Membuat gambar dengan quote yang dipilih

# Menjalankan program
if __name__ == '__main__':
    generate_random_quote_image()
